### DISCUSSION TITLE
Do Leica SL APO lenses need geometric distortion correction?

### DISCUSSION THREAD

**Discussion by x3x7** [lensfun/lensfun#2346] [Q&A] [answered]
Could you please clarify if Leica SL APO lenses really need geometry correction?

I noticed that the lensfun database supports the 35mm and 50mm APO lenses.

I used the formula from here:  https://lensfun.github.io/manual/latest/group__Lens.html#gaa505e04666a189274ba66316697e308e

and parameters from here https://github.com/lensfun/lensfun/blob/master/data/db/mil-leica.xml

To my surprise the deviation from an identity function is very small, as in the picture attached. The input variable is spanning the 0 to 1 interval. Is there a different scale that must be used?
![50mmAPO](https://github.com/user-attachments/assets/6c2ef81b-5ed9-4c09-b3ec-bcd9537e1ed5)

**Comment by mawob** [comment]
With "input variable" you are referring to the radius r, right?
This is the normalized radius (the radius of the largest circle that fits into the image). For 24x36mm (fullframe) this corresponds to a circle with 12mm radius. However, the corners have a distance of approx 22mm from the center, which is ~1.8 times the previous radius. So, the "input variable" r has a range from 0 - 1.8.
Comment: you see correctly that the correction at r=1 (at the right side of your plot) is zero, because this is how it is defined. But you should extend the plot to r=1.8 - that's where the really interesting effects occur.
In this figure, I have made the same plot, but with the axes re-labeled to show the actual radius (in mm along the x-axis) and the radial shift (in microns) for a full frame sensor, for a series of 135mm lenses that all have some amount of pincushion distortion.
![distort-135](https://github.com/user-attachments/assets/434f7cdd-04a5-478d-ab31-d90c6461b1df)
You see that the effects are small between 0-12mm (=the range that your plot is showing) but they become significantly larger towards 22mm.

**Comment by mawob** [comment]
Since I have the plotting code at hand, I quickly made this comparison of the two Leica lenses to some older Carl Zeiss Jena and Voigtlander lenses.
![distort-leica](https://github.com/user-attachments/assets/0bde358b-4d42-46f0-96f6-52dfe6a6d9c9)
These are all pretty good lenses, with relatively low distortion. And you see that the 50mm Leica has almost no distortion at all - and that the Leica 35mm is a little better than most of the others.

**Comment by x3x7** [comment]
Thank your for clarification and all the plots, much appreciated!

Indeed, the input is the undistorted, corrected radius, as mentioned at the formula link. I thought it is normalized with respect to the bigger circle (that contains within it the full image). As such, indeed, the bigger distortion is near the upper limit (1.8) of the input. I have now the results in the plot attached. What is the multiplier to get the microns scale for the Y-axis? 

Even with the updates these APO lenses look to behave quite well if not corrected. In comparison, the Leica Q3 43 uses also an APO lens but with clearly bigger geometric distortion (https://www.dpreview.com/reviews/leica-q3-43-review#IQ). The problem with geometric distortion correction, as far I understand it, is that correcting it comes at some resolution expense. I presume it is better left not corrected if one wants the best resolution.

![dlens](https://github.com/user-attachments/assets/dad1998f-587c-4221-b199-54c9f8ae552f)

**Reply by mawob** [reply, marked as answer]
For a full frame sensor, a normalized radius of unity corresponds to 12 mm. And 1.000 microns is 1 mm.
So, the conversion factor is 12.000.
When you do the distortion correction, the picture is re-sampled. And, technically/conceptually, this leads to some loss.
However, I think that this is completely negligible for all practical purposes (the corresponding effects are smaller as compared to resizing your picture in the jpg/png export).

**Comment by x3x7** [comment]
Regarding the resolution loss with re-sampling I have examined the Leica Q3 43 test. Luckily, Leica does the right thing and does not do geometric corrections in the raw image. However, the raw processing software from dpreview does the correction automatically, not sure if it can even be disabled.

The full image is this:
![L1030016_x4](https://github.com/user-attachments/assets/c327470d-0471-489b-98ab-5d13278b728f)


and the examined area is this:
![L1030016_detail](https://github.com/user-attachments/assets/ed5ae941-3c39-4657-b365-ed2e253eab44)
    
    
If comparing the target area with the processing with distortion correction from dpreview (https://www.dpreview.com/reviews/leica-q3-43-review#IQ) the difference should be quite obvious.
